mypy/checker.py

```diff
@@ -170,52 +170,54 @@ def check_first_pass(self) -> None:
 
         Deferred functions will be processed by check_second_pass().
         """
-        self.errors.set_file(self.path, self.tree.fullname())
-        with self.enter_partial_types():
-            with self.binder.top_frame_context():
-                for d in self.tree.defs:
-                    self.accept(d)
+        with experiments.strict_optional_set(self.options.strict_optional):
```

Comment:
Hm, the -w flag is flawed -- it doesn't show review comments, and it's not sticky.